presto-main/src/test/java/com/facebook/presto/operator/scalar/TestJsonFunctions.java

```diff
@@ -115,6 +122,12 @@ public void testJsonArrayContainsDouble()
     }
 
     @Test
+    public void testJsonExtract()
+    {
+        assertFunction("JSON_EXTRACT_SCALAR(UTF8(X'00 00 00 00 7b 22 72 22'), '$.x')", VARCHAR, null);
```

Comment:
There is a class called TestJsonExtract. Move the test there. And call the method something like testNoAutomaticEncodingDetection. Add comment that with automatic encoding detection, this will be treated as UTF-32.